lib/resources/entity.js

```diff
@@ -0,0 +1,116 @@
+
+
+let alpaca_quote_mapping = {
+    "T": "symbol",
+    "X": "askexchange",
+    "P": "askprice",
+    "S": "asksize",
+    "x": "bidexchange",
+    "p": "bidprice",
+    "s": "bidsize",
+    "c": "conditions",
+    "t": "timestamp"
+};
+
+let alpaca_trade_mapping = {
+    "sym": "symbol",
+    "i": "tradeID",
+    "x": "exchange",
+    "p": "price",
+    "s": "size",
+    "t": "timestamp", // in millisecond
+    "z": "tapeID",
+    "c": "conditions",
+};
+// used in websocket with AM.<SYMBOL>
+let alpaca_agg_minute_bar_mapping = {
+    "sym": "symbol",
+    "v": "volume",
+    "av": "accumulatedVolume",
+    "op": "officialOpenPrice",
+    "vw": "vwap",
+    "o": "openPrice",
+    "h": "highPrice",
+    "l": "lowPrice",
+    "c": "closePrice",
+    "a": "averagePrice",
+    "s": "startEpochTime",
+    "e": "endEpochTime",
+};
+
+// used with rest bars endpoint
+let alpaca_bar_mapping = {
+    "t": "startEpochTime",  // in seconds
+    "o": "openPrice",
+    "h": "highPrice",
+    "l": "lowPrice",
+    "c": "closePrice",
+    "v": "volume",
+};
+
+polygon_quote_mapping = {
+    "sym": "symbol",
+    "ax": "askexchange",
+    "ap": "askprice",
+    "as": "asksize",
+    "bx": "bidexchange",
+    "bp": "bidprice",
+    "bs": "bidsize",
+    "c": "condition",
+    "t": "timestamp"
+};
+
+function AlpacaQuote(data) {
+    let obj = {};
+    for (let [key, value] of Object.entries(data)) {
+        if (alpaca_quote_mapping.hasOwnProperty(key)) {
+            obj[alpaca_quote_mapping[key]] = value;
+        } else {
+            obj[key] = value;
+        }
+    }
+    return obj;
+}
+
+function AlpacaTrade(data) {
```

Comment:
Looks like you don't need to repeat the same pattern but just have curried functions. Just an idea.

Reply:
done